docs/DeveloperGuide.md

```diff
@@ -154,6 +154,117 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+    </div>
+
+6. In `ParserUtil#parseStudentId(String studentId)`, the supplied argument will be trimmed using `String#trim()`.
+7. `StudentId#isValidId(String studentId)` will then be invoked, which checks if the trimmed argument is valid (according to the Regex supplied). If the argument is valid, a new `StudentId` object will be created and returned to the `MarkCommandParser`. If the argument is not valid, a `ParseException` will be thrown.
+8. The `MarkCommandParser` will pass the index input (found in the `ArgumentMultiMap`) into `ParserUtil#parseIndex(Index index)`.
+    <div markdown="span" class="alert alert-info">:information_source:
+    **Note:** A NullException will be thrown if the supplied string argument is null.
+    </div>
+
+9. In `ParserUtil#parseStudentId(Index index)`, the supplied argument will be trimmed using `String#trim()`.
+10. `Index#isValidId(Index index)` will then be invoked, which checks if the trimmed argument is valid (according to the Regex supplied). If the argument is valid, a new `Index` object will be created and returned to the `MarkCommandParser`. If the argument is not valid, a `ParseException` will be thrown.
+11. The `MarkCommandParser` will create a new `MarkCommand` object using the `StudentId` and `Index` created in Step 7 and 10 respectively.
+12. The `LogicManager` will then call `MarkCommand#execute(Model model)`.
+13. The `MarkCommand` will call `model.markTaskOfPerson(Student studentId, Index index)`, which marks the task (corresponding to the supplied index) of the given student as done.
+    <div markdown="span" class="alert alert-info">:information_source:
+    **Note:** A CommandException will be thrown if the supplied index or studentId is invalid, or if the task is already marked as completed.
+    </div>
+    
+14. Lastly, the `MarkCommand` will create a new `CommandResult`, which will be returned to `LogicManager`
+
+The following sequence diagrams shows how the mark command works:
+
+![MarkCommandSequenceDiagram](images/MarkCommandSequenceDiagram-1.png)
+
+![MarkCommandSequenceDiagram](images/MarkCommandSequenceDiagram-2.png)
+
+### Unmark Command
+
+#### Description
+
+The `unmark` command marks a specific done task as undone for a particular student. Since all the fields are compulsory during the execution of the `unmark` command, the user's input is being parsed in `AddressBookParser`. After which, a new `UnmarkCommand` object will be created, and is executed by the `LogicManager`.
+
+#### Implementation
+1. Upon receiving the user input, the `LogicManager` starts to parse the given input text using `AddressBookParser#parseCommand()`.
+2. The `AddressBookParser` invokes the respective `Parser` based on the first word of the input text.
+3. Since the first word in the user input matches the word "unmark", `AddressBookParser#parseCommand(arguments)` will be called. In this case, the arguments refer to the remaining input text after the exclusion of the command word "unmark".
```

Comment:
```suggestion
3. Since the first word in the user input matches the word "unmark", `UnmarkCommandParser#parse(arguments)` will be called. In this case, the arguments refer to the remaining input text after the exclusion of the command word "unmark".
```